### DISCUSSION TITLE
Change Attribute Set

### DISCUSSION THREAD

**Discussion by Danieleeffe1** [OpenMage/magento-lts#3543] [Extensions & Modules]
I would like to point out this interesting plugin for Magento 1.

For those who don't already know it, it is a plug-in that allows the migration of a product from one set of attributes to another.

https://github.com/flagbit/Magento-ChangeAttributeSet

do you think it could be something useful to integrate into OpenMage?

**Comment by addison74** [comment]
The Magento team has not provided the feature to change the attribute set of a product for good reasons. However, there are cases when it would be useful. for example, we have products that did not yet have a well-established set of attributes. We started with a set of skeleton attributes, later the set of attributes was changed directly in the database as number.

I personally agree with an OpenMage to which extensions/modules can be added than one in which changes are constantly injected. It is easier to manage, if there are problems those extensions can be easily deactivated until they are solved. For example this PR #3451, which already has 120 conversations and 99 commits, it should have been adopted as an extension and not directly in OpenMage.

I tested the Flagbit extension a long time ago and found no problems. Changing a set of attributes must be done with caution because orphaned values may remain in the database. Here it depends on how you take it, it can be beneficial if you come back or not if the change is permanent. It is preferable, as I said, to make the change if the initial set of attributes is contained in the final set of attributes.

There is an extension that deletes orphaned attribute values for a product, but I have never used it.

https://github.com/magento-hackathon/EAVCleaner
https://gist.github.com/dfelton/042e15a51b9795b25cad0235a1bf9122

**Reply by sreichel** [reply]
@ADDISON74 Flagbit deletes orphan records too. (there's a config setting to enable/disabe it)

**Reply by empiricompany** [reply]
> For example this PR https://github.com/OpenMage/magento-lts/pull/3451, which already has 120 conversations and 99 commits, it should have been adopted as an extension and not directly in OpenMage.

@ADDISON74 - Sorry, but I have to write here my disappointment about your sentence, since you mentioned my PR.
All new features should be an extension in your opinion? Should we just focus on bug fixes? Is this the future of OpenMage like the old Magento 1 LTS?
Also this PR https://github.com/OpenMage/magento-lts/pull/3220 have 128 onversations and 55 commits, is this the metric to decide if it's ok as PR or should be implemented as an extension?

I remember you wrote me that there was another PR to adjust. I can't find your message, maybe because you deleted it, but i think was this:
https://github.com/OpenMage/magento-lts/pull/2352 - https://github.com/OpenMage/magento-lts/pull/2317
So, also this one should be developed as an extension? Who decides here which PR can be opened and which cannot?

I think there needs to be more respect for everyone's work here and if has reached 120 conversations, maybe it's also due to your discouraging attitude,. You could simply ignore my PR instead of speaking bad of my work, perhaps someone else is interested.
I just wasting my time these days? This attitude can only drive away new developers and limit the community.

- - - 


> do you think it could be something useful to integrate into OpenMage?

@Danieleeffe1 - Regarding the extension, I also use it sometimes, but to integrate it into the core, it would need to be developed and deeply tested. My concern is also that it may leave orphaned data in the database and could create issues.

IMHO it's better to fork and mantain it then list it here: https://github.com/fballiano/awesome-openmage

**Reply by sreichel** [reply]
@empiricompany i have not replied to your last question yet. You did a great job, but I would prefer it as an extension.

From my knowledge the only extension that has been added is N98_LayoutHelper. I'm okay with adding really small, often used extensions. Bigger changes should stay in an extension.

I am one of the flagbit maintainer and can guarantee that it's battle tested over years and has no issues.

**Reply by addison74** [reply]
@empiricompany - the way you replied me proves you did not understand what I meant. There is a lot of anger in your words that you have accumulated since I made some comments in your PR. Please note that those who have done something for this project over the years have respected each other. This mutual respect keeps OpenMage alive after more than 3 years since the official end of Magento. Regarding my opinions about what OpenMage should be, I consider I have left enough marks in this project, without the need to justify myself.

A PR that has over 100 conversations proves a lack of planning, those two "must-have" PRs you mentioned ended precisely because all wanted to do everything in one place. I hope you agree me that if your PR will be integrated in OpenMage, it is not excluded that issues will arise over a while. For this reason I expressed my opinion that I would have preferred it as an extension and not included in the core. It is easier to find the cause of trouble by disabling extensions than debugging the entire code.

This honest public opinion should not demotivate you. I appreciate your work and I found certain approaches useful. I assure you that you can take your work quietly as far as you want because I will not disturb you in any way from now on.

**Reply by addison74** [reply]
@sreichel - I appreciate the tip. I haven't used the extension for a long time, but you gave me a good reason to do it.

**Reply by addison74** [reply]
Related to what should be in the core and what not. This extension is a "must-have" that any OpenMage/Magento store should have installed.

https://github.com/fballiano/openmage-image-cleaner

I had the opportunity to express my comments and I appreciate that the author adopted them, an aspect that gave me the confidence to remove what I was using until then (ImaClean) and put his extension to work. Everything it offers should exist in OpenMage, I even had discussions with several people on this topic, but the arguments I received were in favor of the extension. I respected the opinions and the discussion remained closed.

**Reply by sreichel** [reply]
Everything that can be done with events/observers has no need to be added to core (eg this one, lock attributes)

**Reply by empiricompany** [reply]
@ADDISON74 - That extension by @fballiano is useful and great even though I use a cronjob which run a n98-magerun plugin to remove orphaned images. However, since we can work on the core, we should solve the problem at its root by properly deleting the images when saving / deleting the product.
PS: Another problem is also removing all generated in cache, but there is no way to identify them. The only way is to flush entire cache and it's a problem.

**Reply by fballiano** [reply]
I think it depends on the feature, some should be in the core (advanced grids) some not (like my extension), this is my opinion, depends on the domain.

about my extension. I'll check if the cache thing, if you have more info please tell me.

**Reply by sreichel** [reply]
I think the improvement to grids are to big for adding to core "in time".

EnhancedGrid extension was great. I would highly appreciate a fork and continue work on it.

**Reply by empiricompany** [reply]
Meanwhile I added it here, @fballiano please check this PR:
https://github.com/fballiano/awesome-openmage/pull/1
PS: I think we should create an official awesome-openmage, or better put all in our website

**Reply by fballiano** [reply]
@empiricompany check the new version of the imagecleaner if it solves your issue

**Reply by fballiano** [reply]
@empiricompany the repo is in my name because to make it official we need an RFC, I started https://github.com/OpenMage/rfcs/pull/9 but at the moment it went nowhere.

**Comment by addison74** [comment]
Maybe an RFC should be created to discuss some aspects such as what can be included in the OpenMage core and what not, when to create a PR and when an extension.

I remain of the opinion that any PR that is proposed must be as easy to follow as possible, an aspect that makes it easier to adopt. The more complex it is, the more facilities, the more interest in it will be lost over time and only a few people will have the patience to go through it.

**Reply by empiricompany** [reply]
I agree, the choice of what to include in the core is wide-ranging and depends on each individual's philosophy, which, as already explained, can be more conservative or more forward-thinking. Therefore, we should first understand what the future of the OpenMage project will be: whether it will involve in the development of new features while maintaining backward compatibility as much as possible, as originally planned, or if it will focus on long-term support by providing security patches, bug fixes, etc.
Let's discuss it openly, everyone should voice their opinion!
